### ISSUE TITLE
Merge Event Breakdown

### ISSUE THREAD

**Issue by mattcollier** [gannan08/rdf-canonize-rs#13]
TLDR; `hash_first_degree_quads` is a good target for optmizations.  I'll be doing another breakdown for that function.

### 90 micros
https://github.com/gannan08/rdf-canonize-rust/blob/main/src/urdna2015.rs#L58-L97

#### 3 micros
https://github.com/gannan08/rdf-canonize-rust/blob/main/src/urdna2015.rs#L61-L68

#### 82 micros
https://github.com/gannan08/rdf-canonize-rust/blob/main/src/urdna2015.rs#L83-L89

##### Multiple passes through `hash_first_degree_quads` looks like this (don't mind the rounding)
https://github.com/gannan08/rdf-canonize-rust/blob/main/src/urdna2015.rs#L246-L340

```
FIRST_DEGREE_QUADS_1 39
FIRST_DEGREE_QUADS_1 61
FIRST_DEGREE_QUADS_1 1
FIRST_DEGREE_QUADS_1 6
FIRST_DEGREE_QUADS_1 1
FIRST_DEGREE_QUADS_1 10
```

### 4 micros
https://github.com/gannan08/rdf-canonize-rust/blob/main/src/urdna2015.rs#L98-L119

### 6 micros
https://github.com/gannan08/rdf-canonize-rust/blob/main/src/urdna2015.rs#L174-L236
